CO-1133 let user manage allowing and blocking addresses in Carbonio CE

mbarto opened the pull request:
This PR adds two new sections to the Mails settings-view, to manage lists of allowed and blocked addresses.

:warning: depends from:
 - https://github.com/zextras/carbonio-mailbox/pull/524
 - https://github.com/zextras/carbonio-shell-ui/pull/451

mbarto committed:
refactor: renamed domainWhitelistSubsection into trustedAddressesSubSection

mbarto committed:
feat: added Allowed senders list item to mail subsections

mbarto committed:
feat: added senders-list component

gnekoz committed:
test: add tests for settings subsections

gnekoz committed:
feat: add blocked senders subsection

mbarto committed:
feat: added allowed and blocked address lists to settings-view

mbarto committed:
chore: moved tooltip under the title in senders-list

mbarto committed:
chore: extracted getMessage function

mbarto committed:
chore: extracted type ListType

mbarto committed:
test: added some tests for senders-list

mbarto committed:
feat: added widget for adding email to senders-list

mbarto committed:
test: added add widget test for blocked list

mbarto committed:
feat: added behaviour for add button in senders-list

mbarto committed:
feat: added settings update on add sender in senders-list

mbarto committed:
feat: added settings read in senders-list

mbarto committed:
test: added test for updateSettings with list not empty for senders-list

mbarto committed:
test: clean input after add in senders-list

mbarto committed:
feat: added email validation on add in senders-list

mbarto committed:
refactor: inverted logic of isInvalid to isAddEnabled in senders-list

gnekoz committed:
test: add tests for getSettingsSubSections

mbarto committed:
chore: review of input error messages in senders-list

mbarto committed:
feat: added list of addresses in senders-list

mbarto committed:
feat: implemented address removal in senders-list

mbarto committed:
fix: test on item removal in senders-list

mbarto committed:
chore: removed extra dividers

gnekoz committed:
refactor(trustee-list-item): rename file and component, add data-testid, fix typization, change export

refactor(trustee-addresses): change DS list component

gnekoz committed:
feat: use SendersListItem component and tests

gnekoz committed:
refactor: merge conflicts

gnekoz committed:
fix: remove useless component

gnekoz committed:
chore: remove unused store

gnekoz committed:
chore: inline method and code reformat

gnekoz committed:
chore: remove duplicated test

gnekoz committed:
test: parametrize test

gnekoz committed:
chore: remove useless async keyword

mbarto committed:
feat: switched reading amavis attributes from get-info attrs

mbarto committed:
chore: renamed settingsObj into currentPrefs

mbarto committed:
feat: extended array support in differenceObject to amavis related attributes

mbarto committed:
feat: added support for attributes saving / reading in settings-view

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/zextras/carbonio-mails-ui?pullRequest=591) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/zextras/carbonio-mails-ui?pullRequest=591) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: gnekoz<br/>:x: mbarto<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/zextras/carbonio-mails-ui?pullRequest=591) it.</sub>

mbarto committed:
Merge branch 'devel' into CO-1133-carbonio-ce-let-user-manage-allowing-and-blocking-addresses

# Conflicts:
#	src/views/settings/settings-view.tsx

mbarto committed:
Merge branch 'devel' into CO-1133-carbonio-ce-let-user-manage-allowing-and-blocking-addresses

# Conflicts:
#	translations/en.json

mbarto committed:
fix: senders list can arrive as a single string from the backend

mbarto committed:
feat: introduced saveSettings to avoid using the shell-ui editSettings functionality

mbarto committed:
Merge branch 'devel' into CO-1133-carbonio-ce-let-user-manage-allowing-and-blocking-addresses

# Conflicts:
#	src/views/settings/settings-view.tsx

mbarto committed:
chore: aligned code to work with shell-ui 7.0.x

mbarto committed:
chore: completed save settings with account store update

mbarto committed:
fix: disable save button after successful save

matteobaglini committed:
chore(idea): load prettier configuration

matteobaglini committed:
refactor: extract updateAccountStore function

matteobaglini committed:
refactor: use more specific name for amavis attributes request function

matteobaglini committed:
feat: update settings.attrs property after save

matteobaglini committed:
refactor: make updateAccountStore void

matteobaglini committed:
refactor: properly handle promise status after save with then and catch

nubsthead committed:
wip: updateSettings from shell

matteobaglini committed:
wip: updateAccount from shell

gnekoz committed:
refactor(settings): update Shell settings/account store

gnekoz committed:
refactor: update import method for updateSettings and updateAccount

- Changed the import method for better readability and maintainability
- Ensured compatibility with the existing codebase

gnekoz committed:
Merge remote-tracking branch 'origin/devel' into CO-1133-carbonio-ce-let-user-manage-allowing-and-blocking-addresses

gnekoz committed:
refactor: remove unused code when generating API request to modify the identities

nubsthead committed:
refactor(settings): update saveSettings type

gnekoz committed:
Merge branch 'CO-1133-carbonio-ce-let-user-manage-allowing-and-blocking-addresses' of github.com:zextras/carbonio-mails-ui into CO-1133-carbonio-ce-let-user-manage-allowing-and-blocking-addresses

matteobaglini committed:
refactor(settings): remove no more useful MailMods type

gnekoz committed:
refactor(type): change Shell types import

mbarto committed:
refactor: copied some types from shell that are not exported anymore and updated UpdateAccount type

mbarto committed:
feat: added labels for empty table and for address conflict to senders-list

mbarto committed:
chore: improved style of conflict label in senders-list

gnekoz committed:
build(dep): update Shell dependency

gnekoz committed:
refactor: align tests with the new SoapFault type

gnekoz committed:
refactor: replace board url with context

mbarto committed:
chore: fixed alignment of text and buttons in senders-list

mbarto committed:
feat: added label with items vs max in senders-list

mbarto committed:
chore: fixed alignment of add button in trusted addresses

mbarto committed:
feat: added check to avoid adding more items than max

mbarto committed:
refactor: review logic to enable / disable inputs and buttons in senders-list

mbarto committed:
fix: uncommented code

mbarto committed:
Merge branch 'devel' into CO-1133-carbonio-ce-let-user-manage-allowing-and-blocking-addresses

mbarto committed:
fix: failing tests due to shell changes

mbarto committed:
Merge branch 'devel' into CO-1133-carbonio-ce-let-user-manage-allowing-and-blocking-addresses

# Conflicts:
#	src/app-utils/register-shell-actions.ts
#	src/app-utils/test/register-shell-actions.test.ts
#	src/carbonio-ui-commons
#	src/types/utils/index.d.ts
#	src/ui-actions/message-actions.tsx
#	src/ui-actions/tests/message-actions.test.tsx
#	src/views/app/detail-panel/edit/edit-view.tsx
#	src/views/app/folder-panel/messages/message-list-item.tsx

mbarto committed:
fix: removed duplicated declaration
